File: algorithm/Python3_version/leetcode1072.flip-columns-for-maximum-number-of-equal-rows.py
class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        d = {}
        s = [' ']*(len(matrix[0]))
        for mat in matrix:
            if mat[0]==0:
                for i in range(len(mat)):
                    mat[i] ^= 1
                    s[i] = '1' if mat[i] else '0'
            else:
                for i in range(len(mat)):
                    s[i] = '1' if mat[i] else '0'
            strs = ''.join(s)
            if strs in d:
                d[strs] += 1
            else:
                d[strs] = 1
        maxcount = 0
        for key in d:
            maxcount = max(maxcount, d[key])
        return maxcount

File: algorithm/leetcode1072.flip-columns-for-maximum-number-of-equal-rows.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
    	unordered_map<string,int> m;
    	string s(matrix[0].size(),' ');
    	for(auto& mat : matrix)
    	{
    		if(mat[0] == 0)
    		{
    			for(int i = 0; i < mat.size(); ++i)
    			{
    				mat[i] ^= 1;
    				s[i] = mat[i] ? '1' : '0';
    			}
    		}
    		else
    		{
    			for(int i = 0; i < mat.size(); ++i)
    				s[i] = mat[i] ? '1' : '0';
    		}
    		m[s]++;
    	}
    	int maxCount = 0;
    	for(auto it = m.begin(); it != m.end(); ++it)
    		maxCount = max(maxCount, it->second);
    	return maxCount;
    }
};